test/errors.cpp

```diff
@@ -76,6 +76,11 @@ void shape_transpose_not_permutation4() {
   transpose<0, 0, 4>(s4);
 }
 
+void shape_transpose_not_permutation5() {
+  shape_of_rank<3> s3;
+  transpose<1, 2>(s4);
```

Comment:
Just to confirm - this gather `{s3.y(), s3.z(), s3.z()}`?

Reply:
This is an error, it's not allowed because 0 is missing. This test is things that don't compile. The test is checking that there is only one error in user code (as opposed to a huge spam of template instantiation issues).

Reply:
Oops, this is under errors. Right.